common/configurable/src/main/java/io/helidon/common/configurable/spi/ExecutorServiceSupplierObserver.java

```diff
@@ -0,0 +1,133 @@
+ * limitations under the License.
+ */
+package io.helidon.common.configurable.spi;
+
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.function.Supplier;
+
+/**
+ * Behavior for observers of the various executor service suppliers.
+ * <p>
+ *     This component identifies suppliers to observers using:
+ *     <ul>
+ *         <li>the supplier itself,</li>
+ *         <li>the supplier category (scheduled, server, ad-hoc), and</li>
+ *         <li>the index of this supplier among suppliers in the same category.</li>
+ *     </ul>
+ *     Further, executor services furnished by the suppliers are identified to observers using:
+ *     <ul>
+ *         <li>the executor service itself, and</li>
+ *         <li>the index of the executor service among those from the same supplier.</li>
+ *     </ul>
+ *     The consuming observers can use this identifying information however makes sense for them.
+ *
+ */
+public interface ExecutorServiceSupplierObserver {
+
+    /**
+     * Makes a supplier known to the observer and returns a supplier context for the supplier to use for future interactions
+     * with the observer.
+     *
+     * @param supplier the executor service supplier registering with the observer
+     * @param supplierIndex unique index across suppliers with the same name
+     * @param supplierCategory supplier category for this supplier
+     *
+     * @return the {@code SupplierObserverContext} for the supplier
+     */
+    SupplierObserverContext registerSupplier(Supplier<? extends ExecutorService> supplier,
+                                             int supplierIndex,
+                                             String supplierCategory);
+
+    /**
+     * Makes a supplier known to the observer and returns a supplier context for the supplier to use for future interactions
+     * with the observer, using the {@link io.helidon.common.configurable.spi.ExecutorServiceSupplierObserver.MethodInvocation}
+     * abstraction for invoking methods to obtain metric values.
+     *
+     * @param supplier the executor service supplier registering with the observer
+     * @param supplierIndex unique index across suppliers with the same name
+     * @param supplierCategory the category of supplier registering (e.g., scheduled, server, thread-pool)
+     * @param methodInvocations method invocation information for retrieving interesting information from the supplier's
+     *                          executor services
+     *
+     * @return the {@code SupplierObserverContext} for the supplier
+     */
+    default SupplierObserverContext registerSupplier(Supplier<? extends ExecutorService> supplier,
+                                                     int supplierIndex,
+                                                     String supplierCategory,
+                                                     List<MethodInvocation> methodInvocations) {
+        return null;
```

Comment:
Is this intentional?

Reply:
It was intentional, originally, but there's not really a good reason to have it that way. This method is used only for the virtual threads environment, but any implementation of the SPI should implement this method as well.

Fixed.